redolog/redolog_manager_master.go

```diff
@@ -0,0 +1,149 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redolog
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	"github.com/uber/aresdb/common"
+	"github.com/uber/aresdb/diskstore"
+	"github.com/uber/aresdb/metastore"
+	metaCom "github.com/uber/aresdb/metastore/common"
+	"github.com/uber/aresdb/utils"
+	"sync"
+)
+
+// Master class to create shard level redolog manager
+type RedoLogManagerMaster struct {
```

Comment:
not sure why we need a master for tracking all redolog managers
we only need a factory class for creating redolog manager

Reply:
1. the resource like kafka consumer is in the master
2. also to avoid 1 table/shard being consumed multiple times
basically we are lack of application context, which we should add in the future. and shard/ livestore has no reference back memstore, shard/livestore.